src/main/java/graphql/nadel/engine/transformation/FieldUtils.java

```diff
@@ -15,4 +23,27 @@ public static String resultKeyForField(Field field) {
         return field.getAlias() != null ? field.getAlias() : field.getName();
     }
 
+    public static Field pathToFields(List<String> path) {
+        Field curField = null;
+        for (int ix = path.size() - 1; ix >= 0; ix--) {
+            Field.Builder newField = Field.newField();
+            if (curField != null) {
+                newField.selectionSet(newSelectionSet().selection(curField).build());
+            }
+            newField.name(path.get(ix));
+            curField = newField.build();
+        }
+        return curField;
+    }
+
+    public static LeafExecutionResultNode getLeafNode(ExecutionResultNode executionResultNode) {
```

Comment:
getFirstLeafNode would be a better name and some JavaDoc saying that would be even better!

Reply:
renamed